src/SearchService.Bussines/Commands/Search/SearchCommand.cs

```diff
@@ -1,239 +1,165 @@
 using LT.DigitalOffice.Models.Broker.Requests.Department;
 using LT.DigitalOffice.Models.Broker.Requests.News;
+using LT.DigitalOffice.Models.Broker.Requests.Office;
 using LT.DigitalOffice.Models.Broker.Requests.Project;
 using LT.DigitalOffice.Models.Broker.Requests.User;
+using LT.DigitalOffice.Models.Broker.Requests.Wiki;
 using LT.DigitalOffice.Models.Broker.Responses.Search;
-using LT.DigitalOffice.SearchService.Models.Dto.Enums;
-using LT.DigitalOffice.SearchService.Models.Dto.Models;
 using LT.DigitalOffice.SearchService.Models.Dto.Requests;
-using LT.DigitalOffice.SearchService.Models.Dto.Responses;
+using LT.DigitalOffice.SearchService.Models.Dto.Response;
 using MassTransit;
 using Microsoft.Extensions.Logging;
 using SearchService.Bussines.Commands.Search.Interfaces;
 
-namespace SearchService.Bussines.Commands.Search
+namespace SearchService.Bussines.Commands.Search;
+
+public class SearchCommand : ISearchCommand
 {
-  public class SearchCommand : ISearchCommand
+  private readonly IRequestClient<ISearchDepartmentsRequest> _rcDepartments;
+  private readonly IRequestClient<ISearchNewsRequest> _rcNews;
+  private readonly IRequestClient<ISearchOfficesRequest> _rcOffices;
+  private readonly IRequestClient<ISearchProjectsRequest> _rcProjects;
+  private readonly IRequestClient<ISearchUsersRequest> _rcUsers;
+  private readonly IRequestClient<ISearchWikiRequest> _rcWiki;
+  private readonly ILogger<SearchCommand> _logger;
+
+  public SearchCommand(
+    IRequestClient<ISearchDepartmentsRequest> rcDepartments,
+    IRequestClient<ISearchNewsRequest> rcNews,
+    IRequestClient<ISearchOfficesRequest> rcOffices,
+    IRequestClient<ISearchProjectsRequest> rcProjects,
+    IRequestClient<ISearchUsersRequest> rcUsers,
+    IRequestClient<ISearchWikiRequest> rcWiki,
+    ILogger<SearchCommand> logger)
   {
-    private IRequestClient<ISearchProjectsRequest> _rcProjects;
-    private IRequestClient<ISearchUsersRequest> _rcUsers;
-    private IRequestClient<ISearchDepartmentsRequest> _rcDepartments;
-    private IRequestClient<ISearchNewsRequest> _rcNews;
-    private ILogger<SearchCommand> _logger;
-
-    #region requests
+    _rcUsers = rcUsers;
+    _rcDepartments = rcDepartments;
+    _rcOffices = rcOffices;
+    _rcProjects = rcProjects;
+    _rcNews = rcNews;
+    _rcWiki = rcWiki;
+    _logger = logger;
+  }
 
-    private async Task<List<SearchResultInfo>> SearchProjectsAsync(string text, List<string> errors)
+  public async Task<SearchResultResponse> ExecuteAsync(string text, SearchFilter filter)
+  {
+    if (!filter.IncludeDepartments &&
+        !filter.IncludeNews &&
+        !filter.IncludeOffices &&
+        !filter.IncludeProjects &&
+        !filter.IncludeUsers &&
+        !filter.IncludeWiki)
     {
-      List<SearchResultInfo> projects = new();
+      filter.IncludeDepartments = filter.IncludeNews = filter.IncludeOffices =
+        filter.IncludeProjects = filter.IncludeUsers = filter.IncludeWiki = true;
+    }
```

Comment:
what's point of this?

Reply:
if the query does not specify the sections for which the search has, then the search is performed in all sections